Scripts/Events: Headless Horseman's Shade attacks

SnapperRy opened the pull request:
**Changes proposed:**

Very basic implementation of the Headless Horseman's Shade attacking Horde and Alliance villages during the Hallow's End event.

Original script from SunwellCore with heavy changes.

It works™, but still has many issues.

**Target branch(es):** 3.3.5

**Tests performed:** tested and working.

**Known issues and TODO list:**
- [ ] Missing spawn for orphans near the matrons.
- [ ] Various gossip/texts/sounds missing/to be implemented.
- [ ] Fire handling is very, very approximative. The sniffs we have aren't enough to get a full overview of the event. And youtube videos are quite lacking, too.
- [ ] Pathing and creature spawns are guessed and could use heavy improvement.
- [ ] Guards should help putting out the fires, too.
- [ ] The shade should float down and cast [Conflagration](http://www.wowhead.com/spell=42380/conflagration) to a random nearby player, while saying some text too.
- [ ] Shade doesn't dismount when landing.

SnapperRy committed:
Script/Event: Headless Horseman's Shade attacks.

Keader commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -63,7 +89,504 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+        }
+    };
+
+    SpellScript* GetSpellScript() const
```
missing override

Keader commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -63,7 +89,504 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                GetCaster()->CastSpell(tgt, SPELL_WATER_SPLASH, true);
+        }
+
+        void Register()
```
missing override

Keader commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -63,7 +89,504 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                    owner->SetAuraStack(SPELL_SMALL_FIRE, owner, 20);
+        }
+
+        void Register()
```
missing override

SnapperRy committed:
Delete by ScriptName instead of spell_id and move spell info correction to disables table.

Keader commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -63,7 +89,504 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+        }
+    };
+
+    AuraScript* GetAuraScript() const
```
missing override

Keader reviewed (changes requested):
Missing overrides

SnapperRy commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -63,7 +89,504 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+        void Reset()
+        {
+            eventStarted = false;
+            _events.ScheduleEvent(EVENT_BEGIN, /*Minutes(30)*/Seconds(5));
```
For testing only.

Keader commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -63,7 +89,504 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+        return true;
+    }
+
+    CreatureAI* GetAI(Creature* pCreature) const
```
override ✌️

SnapperRy committed:
Tacklin' life, one override and some codestyle at a time.

SnapperRy committed:
Moar overrides.

SnapperRy committed:
Remove training fire script (doesn't concern this PR), more codestyle.

SnapperRy committed:
Update.

tkrokli commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+    {
+        PrepareAuraScript(spell_hallows_end_base_fire_AuraScript);
+
+        void HandleEffectPeriodicUpdate(AuraEffect* aurEff)
```
```
/home/travis/build/TrinityCore/TrinityCore/src/server/scripts/Events/hallows_end.cpp:136:53: fatal error: 
      unused parameter 'aurEff' [-Wunused-parameter]
        void HandleEffectPeriodicUpdate(AuraEffect* aurEff)
                                                    ^
```

tkrokli replied:
Since travis-ci seems to stop searching for unused parameters after the first it finds, see if you can spot any other missed unused parameters further down in the script (or fix this one and wait for the next travis-ci result).

tkrokli commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                case 159: // Brill
+                    x = 2195.2f; y = 264.0f; z = 55.62f; o = 0.15f; path = 235435;
+                    break;
+                case 3665: // Falcon Wing Square
```
[Google] Did you mean: [**_Falconwing_** Square](https://www.google.com/?gfe_rd=cr&ei=5rIKWNueI5Or8wftjZ3gBg&gws_rd=cr&fg=1#newwindow=1&q=Falcon+Wing+Square)

tkrokli commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+            eventStarted = false;
+        }
+
+        bool GetData(int32 /*type*/) override
```
```
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(219): 
error C3668: 'npc_costumed_orphan_matron::npc_costumed_orphan_matronAI::GetData': 
method with override specifier 'override' did not override any base class methods

C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(219): 
error C4263: 'bool npc_costumed_orphan_matron::npc_costumed_orphan_matronAI::GetData(int32)': 
member function does not override any base class virtual member function
```

tkrokli commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                }
+            }
+        }
+    };
```
```
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(249):
error C4264: 'uint32 UnitAI::GetData(uint32) const': 
no override available for virtual member function from base 'UnitAI'; function is hidden

C:\projects\trinitycore\src\server\game\AI\CoreAI\UnitAI.h(119): note: see declaration of 'UnitAI'
C:\projects\trinitycore\src\server\game\AI\CoreAI\UnitAI.h(139): note: see declaration of 'UnitAI::GetData'
```

tkrokli commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                {
+                    case QUEST_LET_THE_FIRES_COME_A:
+                    case QUEST_LET_THE_FIRES_COME_H:
+                        if (!creature->AI()->GetData())
```
```
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(285):
error C2660: 'UnitAI::GetData': function does not take 0 arguments
```

tkrokli commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                        break;
+                    case QUEST_STOP_THE_FIRES_A:
+                    case QUEST_STOP_THE_FIRES_H:
+                        if (creature->AI()->GetData())
```
```
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(290):
error C2660: 'UnitAI::GetData': function does not take 0 arguments
```

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+        PrepareSpellScript(spell_hallows_end_bucket_lands_SpellScript);
+
+        bool handled;
+        bool Load() { handled = false; return true; }
```
ugh

SnapperRy replied:
So that the same bucket cast doesn't give buckets of water to more than one player hit by the spell.

ccrs replied:
then use a cooldown in the bucket if its a npc or check for any other spell involved such as auras, but if I'm correct this will allow only one cast per server start.

Aokromes replied:
[ 39%] /home/travis/build/TrinityCore/TrinityCore/src/server/scripts/Events/hallows_end.cpp:68:14: fatal error: 
      'Load' overrides a member function but is not marked 'override'
      [-Winconsistent-missing-override]
        bool Load() { handled = false; return true; }
             ^
/home/travis/build/TrinityCore/TrinityCore/src/server/game/Spells/SpellScript.h:131:22: note: 
      overridden virtual function is here
        virtual bool Load() { return true; }
                     ^

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                        owner->GetAI()->DoCast(SPELL_SMALL_FIRE);
+
+                if (Aura* aura = owner->GetAura(SPELL_BIG_FIRE))
+                    if (aura->GetStackAmount() < 20)
```
iirc there is a "getaurastackamount" so you can drop one get here

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+            {
+                if (Aura* aura = owner->GetAura(SPELL_SMALL_FIRE))
+                    if (aura->GetStackAmount() < 20)
+                        owner->GetAI()->DoCast(SPELL_SMALL_FIRE);
```
no need to cast through the AI

ccrs replied:
and actually needed since getai can return null

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                    {
+                        case EVENT_BEGIN:
+                            eventStarted = true;
+                            float x, y, z, o;
```
u need brackets in this case as u are declaring variables inside

ccrs commented:
after seeing this I cry even harder when people say (and I quote) Sunwell is the best core in the world t.t

Exodius commented:
@ccrs maybe it has the potential to be but, it is still heavily outdated in comparison to TC. :D

Exodius commented on sql/updates/world/3.3.5/9999_99_99_99_world.sql:
```diff
@@ -0,0 +1,467 @@
+
+DELETE FROM `creature_text` WHERE `entry`=23543;
+INSERT INTO `creature_text` (`entry`, `groupid`, `id`, `text`, `type`, `language`, `probability`, `emote`, `duration`, `sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES
+(23543, 0, 0, "Prepare yourselves, the bells have tolled! Shelter your weak, your young and your old! Each of you shall pay the final sum. Cry for mercy, the reckoning has come!", 14, 0, 100, 0, 0, 11966, 22022, 0, "lol"),
```
`comment` field should contain something other than "lol" I know that it is there simply as a flavor thing and it is not a priority but `Shade of the Horseman World Event` would be a better description.

This is not addressed directly to you alone. I've seen people leaving out this field empty and/or filling them with random things where a couple of extra words would do them better justice.

SnapperRy replied:
That's a leftover I missed during cleanup phase :P not intended. Thanks for the notice.

joschiwald commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                (*itr)->AreaExploredOrEventHappens(QUEST_STOP_THE_FIRES_H);
+                (*itr)->AreaExploredOrEventHappens(QUEST_STOP_THE_FIRES_A);
+                (*itr)->AreaExploredOrEventHappens(QUEST_LET_THE_FIRES_COME_H);
+                (*itr)->AreaExploredOrEventHappens(QUEST_LET_THE_FIRES_COME_A);
```
isn't there a killcredit?

SnapperRy replied:
Nope, there's no required creature kill in quest_template for those quests, sadly.

SnapperRy committed:
Apply suggested fixes.

SnapperRy committed:
Whops.

SnapperRy committed:
Banish eventStarted forever!

SnapperRy committed:
Even variables want some privacy.

SnapperRy committed:
More fixes.

Keader commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,389 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+            if (Unit* owner = GetUnitOwner())
+                if (roll_chance_i(50))
+                    owner->SetAuraStack(SPELL_BIG_FIRE, owner, 20);
+                else
```
travis complain about backets here

SnapperRy replied:
Fixed that an hour ago and thought I pushed, turns out I didn't :P

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                        break;
+                    }
+                    case EVENT_CLEAVE:
+                        me->CastSpell(me->GetVictim(), SPELL_CLEAVE);
```
you can use here DoCastVictim

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                        {
+                            if (Creature* trigger = me->FindNearestCreature(NPC_FIRE_TRIGGER, 50.0f))
+                                if (!trigger->GetAura(SPELL_START_FIRE) && !trigger->GetAura(SPELL_SPREAD_FIRE))
+                                    me->CastSpell(trigger, SPELL_START_FIRE);
```
and DoCast here

ccrs replied:
I mean just `````DoCast(trigger, SPELL_START_FIRE);````

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+            std::list<Player*> players;
+            Trinity::AnyPlayerInObjectRangeCheck checker(me, radius);
+            Trinity::PlayerListSearcher<Trinity::AnyPlayerInObjectRangeCheck> searcher(me, players, checker);
+            me->VisitNearbyWorldObject(radius, searcher);
```
iirc there is a searcher already similar to getcreatures

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                        if (Creature* shadeOfHorseman = me->SummonCreature(NPC_SHADE_OF_HORSEMAN, x, y, z, o, TEMPSUMMON_CORPSE_TIMED_DESPAWN, 10000))
+                        {
+                            shadeOfHorseman->GetMotionMaster()->MovePath(path, true);
+                            shadeOfHorseman->AI()->DoAction(path);
```
AI can return null, weird things can happen :P

tkrokli replied:
Suggestion for null check here is welcome, so this code can be updated.

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+    {
+        npc_costumed_orphan_matronAI(Creature* creature) : ScriptedAI(creature) {}
+
+        void Reset()
```
override

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+            switch (me->GetAreaId())
+            {
+                case 87: // Goldshire
+                    x = -9494.4f; y = 48.53f; z = 70.5f; o = 0.5f; path = 235431;
```
codestyle

tkrokli replied:
Do you mean all coordinates under each other (NewLine after each semicolon)?

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+            _events.ScheduleEvent(EVENT_BEGIN, /*Minutes(30)*/Seconds(5));
+        }
+
+        void GetInitXYZ(float &x, float &y, float &z, float &o, uint32 &path)
```
should make this helper return boolean.
u know, return false in case the path passed is not handled in the switch

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                        GetInitXYZ(x, y, z, o, path);
+                        if (Creature* shadeOfHorseman = me->SummonCreature(NPC_SHADE_OF_HORSEMAN, x, y, z, o, TEMPSUMMON_CORPSE_TIMED_DESPAWN, 10000))
+                        {
+                            shadeOfHorseman->GetMotionMaster()->MovePath(path, true);
```
MovePath has true as default iirc

tkrokli replied:
https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Movement/MotionMaster.h#L164-L165
(should I change this part too?)

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+
+        void DoAction(int32 param) override
+        {
+            pos = param;
```
this should be a setdata, not a DoAction

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+        {
+            switch (pos)
+            {
+                case 235431: x = -9445.1f; y = 63.27f; z = 58.16f; break;
```
codestyle

ccrs replied:
and same thing as the other helper, make it return false as a sanity check

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+
+                            trigger->AI()->EnterEvadeMode();
+                        }
+                        me->StopMoving();
```
kinda useless since u are doing always FinishEvent

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+
+        void MovementInform(uint32 type, uint32 point) override
+        {
+            if (type == POINT_MOTION_TYPE && point == 1)
```
magic number in point id :D

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                me->SetUInt32Value(UNIT_FIELD_FLAGS, 0);
+                me->SetDisableGravity(false);
+                me->RemoveAura(SPELL_HORSEMAN_MOUNT);
+                //me->Dismount();
```
since u are removing the mount with the RemoveAura, remove this comment?

SnapperRy commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                        if (Creature* shadeOfHorseman = me->SummonCreature(NPC_SHADE_OF_HORSEMAN, x, y, z, o, TEMPSUMMON_CORPSE_TIMED_DESPAWN, 10000))
+                        {
+                            shadeOfHorseman->GetMotionMaster()->MovePath(path, true);
+                            shadeOfHorseman->AI()->DoAction(path);
```
AI should never return null if there's an actual AI assigned, which in this case there is :P

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+            Trinity::PlayerListSearcher<Trinity::AnyPlayerInObjectRangeCheck> searcher(me, players, checker);
+            me->VisitNearbyWorldObject(radius, searcher);
+
+            for (std::list<Player*>::const_iterator itr = players.begin(); itr != players.end(); ++itr)
```
what about ````for (Player* player : players)```` ?

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+
+        void JustDied(Unit* killer) override
+        {
+            Talk(5);
```
Magic number :D

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+        {
+            if (type == POINT_MOTION_TYPE && point == 1)
+            {
+                me->SetUInt32Value(UNIT_FIELD_FLAGS, 0);
```
use SetFlag better

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+            events.Reset();
+            if (failed)
+            {
+                Talk(3);
```
Magic number :D

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+
+            }
+
+            if (!UpdateVictim())
```
this should be on top
if u have events that should trigger when out of combat add a event phase, or a boolean or a separate eventmap or w/e u want to xd

SnapperRy replied:
Why? Other scripts do the same when there are parts of the update that should only happen while in combat.

ccrs replied:
which others? not too many and none of the "big ones" xd

ccrs commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                switch (eventId)
+                {
+                    case EVENT_TALK_1:
+                        Talk(0);
```
Magic number :D

tkrokli replied:
true, it is customary to put the creature_text.comment field there as enum. ^^

Kittnz commented:
So from my observations on retail, this event has 4 main events, first the intro when he spawns and does the first pathing. after eh reaches the end point of the first path he get a new path and just does a circle around something. Normal pathing that continues all the time until someone puts out all the fires or fails. Not sure how much time is in between, about 10 min. After he spawns 3 minutes after he does a new path (only onces) and he cast Conflagration to a random nearby player. after the path is over he stops. Then continues the circle path and then waits until the fires are all put out. If people fail to put out the fire he leaves again with a special path again and says Shade of the Horseman yells: Fire consumes! You've tried and failed.  Let there be no doubt, justice prevailed! During the fight he casts Conflagration and on death he throws a pumpkin somewhere. Think it only have about 2 spawn points. Where he throws it spawns the pumpkin gob where you can complete the quest.

Kittnz commented:
@SnapperRy find me at IRC, i got some stuff for this.

SnapperRy committed:
Thanks for the feedback!

tkrokli commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -57,13 +88,390 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+            me->SetDisableGravity(true);
+        }
+
+        void SpellHit(Unit* caster, const SpellInfo* spellInfo) override
```
```
/home/travis/build/TrinityCore/TrinityCore/src/server/scripts/Events/hallows_end.cpp:267:29: fatal error: 
      unused parameter 'caster' [-Wunused-parameter]
        void SpellHit(Unit* caster, const SpellInfo* spellInfo) override
                            ^
```

Aokromes committed:
Update hallows_end.cpp

Aokromes committed:
Update 9999_99_99_99_world.sql

Keader commented:
Conflicting files
src/server/scripts/Events/hallows_end.cpp

Aokromes committed:
Update 9999_99_99_99_world.sql

tkrokli commented:
@SnapperRy : do you have time to fix the file conflict, so the Travis & AppVeyor checks can be restarted?

Aokromes commented:
^^ It fails because file was removed by ariel xD

tkrokli commented:
Hmm. So remove the file altogether from this PR, or is this PR meaningless now? 😕

SnapperRy commented:
Updating soon™

SnapperRy committed:
From ashes, arise!

SnapperRy committed:
Last unused parameter, hopefully.

Keader commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -0,0 +1,437 @@
+
+        void Reset() override
+        {
+            _events.ScheduleEvent(EVENT_BEGIN, /*Minutes(30)*/Seconds(5));
```
this comment?

Keader commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -0,0 +1,437 @@
+
+    struct npc_costumed_orphan_matronAI : public ScriptedAI
+    {
+        npc_costumed_orphan_matronAI(Creature* creature) : ScriptedAI(creature) {}
```
missing a space { }

Keader commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -0,0 +1,437 @@
+            if (spellInfo->Id == SPELL_START_FIRE || spellInfo->Id == SPELL_SPREAD_FIRE)
+            {
+                DoCast(me, SPELL_FIRE_AURA_BASE, true);
+                DoCast(me, SPELL_SPREAD_FIRE, true);
```
DoCastSelf(SPELL_SPREAD_FIRE, true);

Keader commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -0,0 +1,437 @@
+            {
+                me->SetFlag(UNIT_FIELD_FLAGS, 0);
+                me->SetDisableGravity(false);
+                me->RemoveAura(SPELL_HORSEMAN_MOUNT);
```
RemoveAurasDueToSpell(SPELL_HORSEMAN_MOUNT);

Aokromes committed:
Update hallows_end.cpp

Aokromes committed:
Merge branch '3.3.5' into headless_horseman_shade_attacks

Keader committed:
fix travis and little improvements

Keader committed:
Fixing my OCD (codestyle fixes)

Keader commented:
should still have errors in sql
maybe @tkrokli can fix ? =)

Commit:
Update and rename 9999_99_99_99_world.sql to 2017_99_99_99_world.sql

- changed 1 entry to CreatureID in creature_text query
- general column alignment to increase value readability

tkrokli commented:
The SQL file is updated and passes travis-ci check.

STARRHELD commented:
This would be awesome! :)

jackpoz commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -0,0 +1,523 @@
+                    o = 3.4f;
+                    path = 235436;
+                    break;
+            }
```
please add a default case initializing all variables

jackpoz commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -0,0 +1,523 @@
+                    y = -6833.8f;
+                    z = 18.5f;
+                    break;
+            }
```
please add a default case initializing all variables

Commit:
Add default case with initialize to both switches

-  as requested

tkrokli commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -0,0 +1,534 @@
+public:
+    npc_soh_fire_trigger() : CreatureScript("npc_soh_fire_trigger") { }
+
+    struct npc_soh_fire_triggerAI : public NullCreatureAI
```
```
src/server/scripts/Events/hallows_end.cpp:180:44: fatal error:
 unknown class name 'NullCreatureAI'; did you mean 'CreatureAI'?

    struct npc_soh_fire_triggerAI : public NullCreatureAI
                                           ^~~~~~~~~~~~~~
                                           CreatureAI

src/server/game/Entities/Creature/Creature.h:31:7
Building CXX object src/server/scripts/CMakeFiles/scripts_commands.dir/Commands/cs_ahbot.cpp.o:
 note: 'CreatureAI' declared here
class CreatureAI;
      ^

1 error generated.
```

tkrokli replied:
Why wasn't this error triggered before? Recent changes?

joschiwald replied:
#include "PassiveAI.h"

Commit:
include PassiveAI.h in hallows_end.cpp

tkrokli commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -0,0 +1,535 @@
+        {
+            float radius = 100.0f;
+            std::list<Player*> players;
+            Trinity::AnyPlayerInObjectRangeCheck checker(me, radius);
```
```
src/server/scripts/Events/hallows_end.cpp:395:49:
 fatal error: expected ';' after expression

            Trinity::AnyPlayerInObjectRangeCheck checker(me, radius);
                                                ^
                                                ;
1 error generated.
```
That can not be a correct interpretation by travis-ci - I guess something else needs to be fixed?
From what I can find in search results, that code is used elsewhere in the code without triggering this error.

tkrokli replied:
Oh, I see. AppVeyor has got a more detailed explanation.
```
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(395):
 error C2039: 'AnyPlayerInObjectRangeCheck': is not a member of 'Trinity' [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
  C:\projects\trinitycore\src\server\game\Spells\Auras\SpellAuraEffects.h(322):
  note: see declaration of 'Trinity'

  C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(395): error C2065:
 'AnyPlayerInObjectRangeCheck': undeclared identifier [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]

C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(395): error C2146:
 syntax error: missing ';' before identifier 'checker' [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]

C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(395): error C3861:
 'checker': identifier not found [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]

 C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(396): error C2039: 'PlayerListSearcher': is not a member of 'Trinity' [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
  C:\projects\trinitycore\src\server\game\Spells\Auras\SpellAuraEffects.h(322): note: see declaration of 'Trinity'
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(396): error C2065: 'PlayerListSearcher': undeclared identifier [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(396): error C2039: 'AnyPlayerInObjectRangeCheck': is not a member of 'Trinity' [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
  C:\projects\trinitycore\src\server\game\Spells\Auras\SpellAuraEffects.h(322): note: see declaration of 'Trinity'
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(396): error C2065: 'AnyPlayerInObjectRangeCheck': undeclared identifier [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(396): error C2065: 'checker': undeclared identifier [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(396): error C3861: 'searcher': identifier not found [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]

C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(397): error C2039: 'VisitNearbyWorldObject': is not a member of 'Creature' [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
  C:\projects\trinitycore\src\server\game\Spells\SpellScript.h(31): note: see declaration of 'Creature'
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(397): error C2065: 'searcher': undeclared identifier [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
```

tkrokli replied:
OK, so maybe this needs another include, like
```cpp
#include "GridNotifiers.h"
```

Commit:
added include GridNotifiersImpl.h

tkrokli commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -0,0 +1,536 @@
+            std::list<Player*> players;
+            Trinity::AnyPlayerInObjectRangeCheck checker(me, radius);
+            Trinity::PlayerListSearcher<Trinity::AnyPlayerInObjectRangeCheck> searcher(me, players, checker);
+            me->VisitNearbyWorldObject(radius, searcher);
```
```
src/server/scripts/Events/hallows_end.cpp:398:17:
 fatal error: no member named 'VisitNearbyWorldObject' in 'Creature'
            me->VisitNearbyWorldObject(radius, searcher);
            ~~  ^
1 error generated.
```
I would think this needs another `#include` , but I can't find `VisitNearbyWorldObject()` in the source at all.

tkrokli replied:
Meh, it has been changed to `VisitNearbyObject` about the time this PR was started. :-P

Commit:
VisitNearbyObject

- VisitNearbyWorldObject changed to VisitNearbyObject

Commit:
Catching up with 2017

- looked up commit d6201e5dbb7f (2017-05-13)

jackpoz commented:
@Keader could you check if your review was taken care of?
@SnapperRy and @ccrs should the Known Issues in the PR description be addressed before merging this PR ?

joschiwald commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -0,0 +1,537 @@
+            Talk(SAY_DEFEATED);
+            float x, y, z;
+            GetPosToLand(x, y, z);
+            me->CastSpell(x, y, z, SPELL_SUMMON_LANTERN, true);
```
this will never work, the spell requires a nearby unit target

joschiwald commented on src/server/scripts/Events/hallows_end.cpp:
```diff
@@ -0,0 +1,537 @@
+        {
+            if (type == POINT_MOTION_TYPE && point == POINT_HORSEMAN_ATTACK)
+            {
+                me->SetFlag(UNIT_FIELD_FLAGS, 0);
```
this line will do nothing, if you want to remove some flags, then remove it

jackpoz commented:
could anyone update this PR to look good enough to be merged please ? :)

illfated commented:
if you want the existing PR updated, I presume you are asking some of the other TC members with write access and knowledge to make the needed changes.

On the off chance you are asking anyone for assistance, I could fork the PR branch and open a new PR, but with knowledge limited to fixing the already explained required changes.

Kittnz commented:
This is incomplete also wrong waypoints, perhapse i can take a look again into this.

jackpoz commented:
@Kittnz how did it go :) ?

jackpoz commented:
Closing this PR as it seems abandoned. Feel free to ask here to reopen it if anyone would like to work on it.

offl commented:
Another implementation https://github.com/cmangos/mangos-tbc/commit/29ec5b4bc1eb228fa437c430c412a7b5d47d47fd
